src/compiler/checker.ts

```diff
@@ -2225,8 +2225,8 @@ namespace ts {
                             !checkAndReportErrorForExtendingInterface(errorLocation) &&
                             !checkAndReportErrorForUsingTypeAsNamespace(errorLocation, name, meaning) &&
                             !checkAndReportErrorForExportingPrimitiveType(errorLocation, name) &&
+                            !checkAndReportErrorForUsingNamespaceAsTypeOrValue(errorLocation, name, meaning) &&

```

Comment:
It’s been a while, but I _believe_ it’s this:

A namespace is a value if it exports values. If it’s not a value, it’s not a type per se, but it’s fairly typey. I believe after tweaking the `meaning` flag checks in these functions to be a bit easier to read, the `checkAndReportErrorForUsingTypeAsValue` started reporting an error on typey (uninstantiated) namespaces, which was stepping on the toes of `checkAndReportErrorForUsingNamespaceModuleAsValue`, which wanted to issue essentially the same error, but slightly more precise in mentioning that the non-valuey thing is a namespace. So I simply moved that check first so it would take precedence. So I actually think this order change _isn’t_ reflected in the tests, because it just neutralizes the effect of another readability change. (I also renamed `checkAndReportErrorForUsingNamespaceModuleAsValue` to `checkAndReportErrorForUsingNamespaceAsTypeOrValue` because nobody knows what a “namespace module” is, and it was already issuing errors when using a namespace as a type.)